### PR TITLE
Gas optimization certora (ETH-1626)

### PR DESCRIPTION
**Pull request by iamsahu**
## Description

Incorporates the gas changes suggested by Certora

## Notice

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/liquid-collective/liquid-collective-protocol/pulls) for the same update/change?
- [x] Have you assigned this PR to yourself?
- [x] Have you added at least 1 reviewer?
- [ ] Have you updated the official documentation?
- [ ] Have you added sufficient documentation in your code?
- [ ] Have you added relevant tests to the official test suite?

## Pull Request Type

- [ ] 💫 New Feature (Breaking Change)
- [ ] 💫 New Feature (Non-breaking Change)
- [ ] 🛠️ Bug fix (Non-breaking Change: Fixes an issue)
- [x] 🕹️ Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)

### CODE REVIEW

File: contracts/test/OperatorsRegistry.1.t.sol [hunk 1]
```diff
@@ -894,48 +894,48 @@ contract OperatorsRegistryV1Tests is OperatorsRegistryV1TestBase, BytesGenerator
...
-        emit RemovedValidatorKey(index, LibBytes.slice(tenKeys, 0, 48));
-        operatorsRegistry.removeValidators(index, indexes);
-        operator = operatorsRegistry.getOperator(index);
-        assert(operator.keys == 5);
-        assert(operator.limit == 5);
-
-        operatorsRegistry.addValidators(index, 10, tenKeys);
-        operator = operatorsRegistry.getOperator(index);
-        assert(operator.keys == 15);
-    }
```
Comment (mischat) [inline contracts/test/OperatorsRegistry.1.t.sol:938]: @iamsahu why did we delete these tests ? I would have imagined that business logic shouldn't have changed in this PR ?

Comment (iamsahu) [reply contracts/test/OperatorsRegistry.1.t.sol]: The test has been left out incorrectly. I have fixed it in the next PR. There was also a bug introduced due to incorrect suggestion from report being implemented.

File: contracts/src/state/operatorsRegistry/Operators.2.sol [hunk 2]
```diff
@@ -143,20 +145,8 @@ library OperatorsV2 {
...
-                unchecked {
-                    ++activeIdx;
-                }
-            }
-            unchecked {
-                ++idx;
-            }
+        assembly ("memory-safe") {
+            mstore(activeOperators, activeCount)
         }
```
Comment (mischat) [inline contracts/src/state/operatorsRegistry/Operators.2.sol:150]: Do we have sufficient tests to ensure that this assembly code is robust ?

Looking at the test file we seem to have 2 that cover listActiveOperators()?

**Review by mischat — COMMENTED** [review]
I have a couple of comments / questions, which we could resolve before closing the PR, thanks @iamsahu

File: contracts/src/state/operatorsRegistry/Operators.2.sol (hunk 2 above)
Comment (mischat) [reply contracts/src/state/operatorsRegistry/Operators.2.sol:150]: @iamsahu do you feel like we have enough test coverage here ?

File: contracts/src/River.1.sol [hunk 3]
```diff
@@ -117,7 +117,9 @@ contract RiverV1 is
         // force committed balance to a multiple of 32 ETH and
         // move extra funds back to the deposit buffer
         uint256 dustToUncommit = CommittedBalance.get() % DEPOSIT_SIZE;
-        _setCommittedBalance(CommittedBalance.get() - dustToUncommit);
+        unchecked {
+            _setCommittedBalance(CommittedBalance.get() - dustToUncommit);
+        }
         _setBalanceToDeposit(BalanceToDeposit.get() + dustToUncommit);
```
[the remark this answers is not in the document]
Comment (mischat) [reply contracts/src/River.1.sol:123]: @iamsahu could you add it in, and then i could resolve the conversation ?

[the remark this answers is not in the document]
Comment (iamsahu) [reply contracts/src/River.1.sol]: This has been added.

File: contracts/src/state/operatorsRegistry/Operators.2.sol (hunk 2 above)
Comment (mischat) [reply contracts/src/state/operatorsRegistry/Operators.2.sol:150]: great, thanks for adding in more tests here

**Pull request merged**